src/main/java/com/blamejared/crafttweaker/api/zencode/impl/native_types/CrTJavaNativeClassConverter.java

```diff
@@ -0,0 +1,89 @@
+package com.blamejared.crafttweaker.api.zencode.impl.native_types;
+
+import com.blamejared.crafttweaker.api.CraftTweakerAPI;
+import com.blamejared.crafttweaker.api.zencode.impl.registry.ZenClassRegistry;
+import com.blamejared.crafttweaker.impl.native_types.CrTNativeTypeInfo;
+import com.blamejared.crafttweaker.impl.native_types.NativeTypeRegistry;
+import org.openzen.zencode.java.ZenCodeType;
+import org.openzen.zencode.java.module.JavaNativeTypeConversionContext;
+import org.openzen.zencode.java.module.converters.*;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.util.Arrays;
+
+class CrTJavaNativeClassConverter extends JavaNativeClassConverter {
+    
+    private final ZenClassRegistry zenClassRegistry;
+    
+    public CrTJavaNativeClassConverter(JavaNativePackageInfo packageInfo, JavaNativeTypeConversionContext typeConversionContext, JavaNativeTypeConverter typeConverter, JavaNativeHeaderConverter headerConverter, JavaNativeMemberConverter memberConverter, ZenClassRegistry zenClassRegistry) {
+        super(typeConverter, memberConverter, packageInfo, typeConversionContext, headerConverter);
+        this.zenClassRegistry = zenClassRegistry;
+    }
+    
+    @Override
+    public String getNameForScripts(Class<?> cls) {
+        if(getNativeTypeRegistry().hasInfoFor(cls)) {
+            return getNativeTypeRegistry().getCrTNameFor(cls);
+        }
+        
+        if(cls.getCanonicalName().startsWith("net.minecraft")) {
```

Comment:
Does this work in production?

Reply:
As far as I know, yes, only methods and fields are obfuscated.
ATs use the full class name as well ^^